src/plugins/intel_gpu/src/graph/impls/cpu/eltwise.cpp

```diff
@@ -179,7 +179,7 @@ struct eltwise_impl : public typed_primitive_impl<eltwise> {
 
         auto output_mem_ptr = instance.output_memory_ptr();
 
-        cldnn::mem_lock<uint8_t, mem_lock_type::write> output_lock(output_mem_ptr, stream);
+        cldnn::mem_lock<uint8_t, mem_lock_type::read> output_lock(output_mem_ptr, stream);
```

Comment:
Why? It's output mem, so I think it should be locked for write

Reply:
in new commit I've added read lock because without it write lock can't be created, is it ok?

Reply:
That still looks weird. Could you explain what's the goal? If you need to read the data somewhere, then you may use `read_write` mode

Reply:
I want to have only write lock, but I cant create it because of that: 
![image](https://github.com/openvinotoolkit/openvino/assets/161812866/7cb4a7ad-0149-4010-85c3-e683044232d5)

Reply:
In that case the problem is not related to lock type. Even if you set `read` lock type for USM device memory, it will create a temporary host-accessible memory buffer and copy device memory to it. But once the lock is released, this temporary buffer is freed w/o write back to original USM device buffer. So basically, you won't have correct accuracy with that approach. 

So if you have a USM device memory buffer in CPU impl, you need to check the following:
- Ensure that CPU impl type is expected. For eltwise is can be used for shape flow only (shape of subgraphs).
- Check why out memory has USM Device allocation type. We expect that CPU impls requires lockable memory types (USM host or cl::Buffer) - so probably there's an issue with allocation type selection
- If for some reason USM device is preferrable, then you'll need to create a temporary buffer for this op to write result initially, and then enqueue memcpy to actual out buffer

Reply:
@vladimir-paramuzov thanks for insights, cpu impl is expected because it is flow of shape of subgraphs. usm device alloc for output happened because there is add->concat, and in code when there is concat, output of node(add node) is set to  concat_output
![image](https://github.com/openvinotoolkit/openvino/assets/161812866/6b54e10b-247f-4c12-9f6b-7c6687143cfb)
![image](https://github.com/openvinotoolkit/openvino/assets/161812866/db45e5c4-929b-4d63-95a8-af7b5885d67c)

Reply:
@michal-miotk I think shape_of subgraph should end on Gather operation (i.e. Range and its users are not part of shape flow) - looks like that's the initial issue

Reply:
@michal-miotk , did you check if range primitive should be included into shapeof-subgraph? If it should, maybe we just need to disable concat optimization for nodes which have ShapeOf-subgraph nodes in dependencies

Reply:
I will check if should be included into shapeof-subgraph

Reply:
@vladimir-paramuzov gather node not have compressed weights so can be marked
![image](https://github.com/openvinotoolkit/openvino/assets/161812866/6bf94081-7bdc-4da7-aa1b-c1c69ae97b26)

Reply:
@sshlyapn Shouldn't Range always terminate shape_of subgraphs?

Reply:
@sshlyapn  output shape is:
![image](https://github.com/openvinotoolkit/openvino/assets/161812866/6d6b80ef-6aac-42f6-9d51-0e6ffcaddf61)
disable concat optimization for nodes which have ShapeOf-subgraph nodes in dependencies is enough to make it work

Reply:
pushed